boa/vm/fork.py

```diff
@@ -66,69 +69,55 @@ def get_rpc(cls, url, cache_file=DEFAULT_CACHE_DIR):
         if (url, cache_file) in cls._loaded:
             return cls._loaded[(url, cache_file)]
 
-        ret = cls(url, cache_file)
+        ret = cls(EthereumRPC(url), cache_file)
         cls._loaded[(url, cache_file)] = ret
         return ret
 
     # a stupid key for the kv store
     def _mk_key(self, method: str, params: Any) -> Any:
         return json.dumps({"method": method, "params": params}).encode("utf-8")
 
-    # note: overrides super().fetch!
     def fetch(self, method, params):
         # dispatch into fetch_multi for caching behavior.
         (res,) = self.fetch_multi([(method, params)])
         return res
 
     # caching fetch of multiple payloads
-    # note: overrides super().fetch_multi!
     def fetch_multi(self, payload):
         ret = {}
-        ks = []
+        keys = []
         batch = []
-        for i, (method, params) in enumerate(payload):
-            k = self._mk_key(method, params)
+        for payload_index, (method, params) in enumerate(payload):
+            key = self._mk_key(method, params)
             try:
-                ret[i] = json.loads(self._db[k])
+                ret[payload_index] = json.loads(self._db[key])
             except KeyError:
-                ks.append(k)
-                batch.append((i, method, params))
+                keys.append((key, payload_index))
+                batch.append((method, params))
 
         if len(batch) > 0:
-            res = self._raw_fetch_multi(batch)
-            for i, s in res.items():
-                k = ks[i]
-                ret[i] = s
-                self._db[k] = json.dumps(s).encode("utf-8")
+            for batch_index, rpc_result in enumerate(self._rpc.fetch_multi(batch)):
```

Comment:
and maybe a comment that we need to keep track of the keys/original order because we don't send the cached requests to the rpcs, so we only need to line up the requests that actually get sent.

Reply:
Done!